packages/website/components/enterpriseTierRequest/enterpriseTierRequestModal.js

```diff
@@ -0,0 +1,83 @@
+import { useState } from 'react';
+
+import Modal from 'modules/zero/components/modal/modal';
+import CloseIcon from 'assets/icons/close';
+import Button from 'components/button/button.js';
+import { createEnterpriseTierRequest } from 'lib/api';
+import GradientBackground from 'components/gradientbackground/gradientbackground';
+
+const UserRequestModal = ({ isOpen, onClose }) => {
+  const [requesting, setRequesting] = useState(false);
+
+  async function handleCreateUserRequest(e) {
+    e.preventDefault();
+    const data = new FormData(e.target);
+
+    const authMethod = data.get('auth-method');
+    const links = data.get('links');
+    const dataVolume = data.get('data-volume');
+    const dataReadTypeAndFrequency = data.get('data-read-type-and-frequency');
+    const additionalInfo = data.get('additional-info');
+
+    if (authMethod && links && dataVolume && dataReadTypeAndFrequency) {
```

Comment:
I like the idea of checking these in the js even thought the html has `required`.
I think it can be even better by avoiding a situation where it might 'silently fail', which could make debugging quite a pain if things shift over time (e.g. if someone accidentally removes `required` in the html below).

I'd either throw in the unexpected case
```
if (authMethod && links && dataVolume && dataReadTypeAndFrequency) {
  throw new Error(`cannot build enterprise tier request because one of the required fields is falsy`);
}
```

Or if that feels aggressive, at least add an `else` block to this if with a `console.warn` or similar explaining to a future dev why the normal behavior was not attempted.

Reply:
I like the idea of throwing here.  I didn't put too much thought into this as I was just copying from the existing modal.

Reply:
OK, I think I've fixed it, but I want to leave this comment unresolved until you verify it.